src/pkg/fault/fault.go

```diff
@@ -168,70 +170,12 @@ func (e *Bus) addRecoverableErr(err error) bool {
 
 	e.recoverable = append(e.recoverable, err)
 
-	return isFail
-}
-
-// AddSkip appends a record of a Skipped item to the fault bus.
-// Importantly, skipped items are not the same as recoverable
-// errors.  An item should only be skipped under the following
-// conditions.  All other cases should be handled as errors.
-// 1. The conditions for skipping the item are well-known and
-// well-documented.  End users need to be able to understand
-// both the conditions and identifications of skips.
-// 2. Skipping avoids a permanent and consistent failure.  If
-// the underlying reason is transient or otherwise recoverable,
-// the item should not be skipped.
-func (e *Bus) AddSkip(ctx context.Context, s *Skipped) {
-	if s == nil {
-		return
+	// local bus instances must promote errors to the root bus.
+	if e.parent != nil {
+		e.parent.addRecoverableErr(err)
```

Comment:
We are not acquiring the lock on the parent for this change as that step is done in `AddRecoverableErr`(the public function). I think we should just move the lock acquiring part to the private function as the other thing is printing a log message which does not require a lock anyways.

Reply:
Yup, good call.